### ISSUE TITLE
Experiment: new reward normalization approach

### ISSUE THREAD

**Issue by AdamGleave** [HumanCompatibleAI/imitation#261]
We currently use `VecNormalize` to normalize the scale of the reward over time. However, this has a few problems:
  1. It depends on the reward function *in the past*. So if e.g. the reward function used to have large scale, `VecNormalize` will scale it down, even if the current reward function is of smaller scale.
  2. It depends on the distribution of the generator function.
Both of these make the reward function (more) non-stationary, which would be expected to cause problems, although I'll note 2) applies in the typical case of a fixed reward function and `VecNormalize` does not seem to usually cause problems with RL training in that case.

One simple idea is to collect a fixed set of rollouts (e.g. from a random policy), and then normalize the reward function based on evaluations on that set of fixed rollouts.

I think this is analogous to a crude form of batch normalization. One could also do batchnorm at every layer but on this fixed set.

Self-assigning but I might not have time to do this for many months (if ever), so if someone else feels this is promising, feel free to try it out.

**Comment by AdamGleave** [comment]
This is mostly addressed by the EMA normalization approach.
